### PR TITLE
Unit Tests

### PR DESCRIPTION
**Pull request by ursaciuc-adrian**
Reinitialized the events list at every test.
Separated some classes.

### CODE REVIEW

File: FiiPrezent.Tests/EventsServiceTests.cs [hunk 1]
```diff
@@ -1,40 +1,42 @@
...
+        private EventsService _service;
+
+        [SetUp]
+        public void SetUp()
         {
-            var service = new EventsService();
-            var result = service.RegisterParticipant("bad code", "test participant");
-            result.ShouldBeNull();
+            _service = new EventsService();
+            _service.InitializeEvents();
```
Comment (sdamian) [inline FiiPrezent.Tests/EventsServiceTests.cs:15]: No need to switch to NUnit for this. It's enough to use the ctor with XUnit.

File: FiiPrezent.Tests/EventsServiceTests.cs [hunk 2]
```diff
@@ -1,40 +1,42 @@
...
+        [Test]
+        public void RegisterParticipant_WhenCodeIsInvalid_ReturnNull()
         {
-            var service = new EventsService();
-            var result = service.RegisterParticipant("cometothedarksidewehavecookies", "test participant");
-            result.ShouldNotBeNull();
-            
+            var result = _service.RegisterParticipant("bad code", "test participant");
+
+            Assert.That(result, Is.Null);
```
Comment (sdamian) [inline FiiPrezent.Tests/EventsServiceTests.cs:23]: Shouldly works with any testing framework and does something that the regular asserts do not. It includes the expression on the left side of the "should" in the message (see docs for an example).
Why the change?

File: FiiPrezent.Tests/FiiPrezent.Tests.csproj [hunk 3]
```diff
@@ -10,6 +10,8 @@
     <PackageReference Include="Microsoft.AspNetCore.All" Version="2.0.5" />
     <PackageReference Include="Microsoft.AspNetCore.TestHost" Version="2.0.1" />
     <PackageReference Include="Microsoft.NET.Test.Sdk" Version="15.5.0" />
+    <PackageReference Include="NUnit" Version="3.10.1" />
+    <PackageReference Include="NUnit3TestAdapter" Version="3.10.0" />
```
Comment (sdamian) [inline FiiPrezent.Tests/FiiPrezent.Tests.csproj]: XUnit is actually the (relatively) newer one... Why do you want to switch?

File: FiiPrezent/ViewModels/ErrorViewModel.cs [hunk 4]
```diff
@@ -1,11 +1,9 @@
...
-namespace FiiPrezent.Models
-{
-    public class ErrorViewModel
-    {
-        public string RequestId { get; set; }
-
-        public bool ShowRequestId => !string.IsNullOrEmpty(RequestId);
-    }
-}
+namespace FiiPrezent.ViewModels
```
Comment (sdamian) [inline FiiPrezent/ViewModels/ErrorViewModel.cs:1]: I'm not sure about the moving these to the "ViewModels" folder. The convention in MVC is that the (view) models stay in the "Model" folder :)

File: FiiPrezent/Services/EventsService.cs [hunk 5]
```diff
@@ -17,22 +17,35 @@ public class EventsService
...
+        public Event RegisterParticipant(string code, string name)
+        {
+            var @event = FindEventByVerificationCode(code);
+            @event?.RegisterParticipant(name);
 
             return @event;
         }
 
-        public Event FindEventByVerificationCode(string modelCode)
+        public Event FindEventByVerificationCode(string code)
```
Comment (sdamian) [inline FiiPrezent/Services/EventsService.cs:41]: Good catch 👍

File: FiiPrezent/Services/EventsService.cs [hunk 6]
```diff
@@ -17,22 +17,35 @@ public class EventsService
...
-            var @event = FindEventByVerificationCode(verificationCode);
-            @event?.RegisterParticipant(participantName);
+            _events = new List<Event>
+            {
+                new Event
+                {
+                    Name = "Best training ever",
+                    Description = "Modern web application development with ASP.NET Core :o",
+                    VerificationCode = "cometothedarksidewehavecookies"
+                }
```
Comment (sdamian) [inline FiiPrezent/Services/EventsService.cs:29]: Don't duplicate code...

File: FiiPrezent/Services/EventsService.cs [hunk 7]
```diff
@@ -17,22 +17,35 @@ public class EventsService
             }
         };
 
-        public Event RegisterParticipant(string verificationCode, string participantName)
+        public void InitializeEvents()
```
Comment (sdamian) [inline FiiPrezent/Services/EventsService.cs:20]: Using static variables if the real problem here. We'll find a more elegant solution on the next session.

**Review by sdamian — CHANGES_REQUESTED** [review]

File: FiiPrezent.Tests/EventsServiceTests.cs (hunk 2 above)
Comment (ursaciuc-adrian) [reply FiiPrezent.Tests/EventsServiceTests.cs:23]: I am more familiar with this one.

File: FiiPrezent/ViewModels/ErrorViewModel.cs (hunk 4 above)
Comment (ursaciuc-adrian) [reply FiiPrezent/ViewModels/ErrorViewModel.cs:1]: A model is a class representation and has no knowledge of the View (like Event.cs) and a ViewModel is a data binder between the view and the model (EventViewModel.cs). Also, the viewModels contains attributes like DisplayName or Required. This is why I like to separate them. I think this is the MVVM pattern.

Comment (sdamian) [reply FiiPrezent/ViewModels/ErrorViewModel.cs:1]: MVVM is used in frameworks that allow two way binding (Model->View and View->Model hence the name) like WPF. MVC doesn't work that way.
In our case we use different models for GET and POST. The decision to name them ViewModels was mine to emphasise that the service layer should not know about them. This is definitely not MVVM.

File: FiiPrezent.Tests/EventsServiceTests.cs (hunk 2 above)
Comment (sdamian) [reply FiiPrezent.Tests/EventsServiceTests.cs:23]: Not a valid reason to change it.

**Comment by sdamian** [comment]
Please send PR's from a feature branch.. every time you update the PR source branch the PR is also updated. 
"Files changed 175" - is a bit much
